ui/src/present.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright © 2021 The Radicle Upstream Contributors
+//
+// This file is part of radicle-upstream, distributed under the GPLv3
+// with Radicle Linking Exception. For full terms see the included
+// LICENSE file.
+
+function shorten(
+  value: string,
+  length: number = 5,
+  offset: number = 2
+): string {
+  return `${value.slice(0, length + offset)}…${value.slice(-length)}`;
+}
```

Comment:
From the argument names I assumed a different behavior from what this function does. I assumed that `length` was the overall length of the shortened string and that whatever was before `offset` was not included in the output.

In addition, it’s good practice to only have default arguments when the majority of the calls uses a specific value. Making a parameter non-optional forces the user to consider the value which I think is important here because there are so many different use cases. 

In light of this, I would suggest the following
```typescript
function shorten(value, beginningLen: number, endLen?: number) {
  return `${value.slice(0, beginningLen)}…${value.slice(end ?? beginningLen)}`;
}

Reply:
2729e2b2dfbf525262149b184e0f510753e7d3e9